### ISSUE TITLE
Spike Cucumber vs Capybara

### ISSUE THREAD

**Issue by thenet84** [thenet84/walking-fish#2]

**Comment by shen-sat** [comment]
_TLDR:_
Cucumber and capybara are different test frameworks.
It's perfectly possible to write feature tests using only rspec and capybara. Cucumber provides additional methods used to help make the tests more readable for non-devs.
As we don't have to worry about non-devs, I recommend we use only rspec and capybara for testing.
----------
Let’s say we have a cat website, and on the homepage we have the text “Welcome to the cat website”. To test that this text actually appears on the homepage, you would normally need to visit the page and read the text yourself. Or you could write a (feature) test to do this for you. This is where capybara comes in. 

Capybara can spin up a browser, visit a website, interact with it using methods like `fill_in` and `click_on` and then it can return page content so that you can check it against what you expect the content to be (using rspec). For example, a test using capybara for the above scenario would be:
```
RSpec.describe ‘Visit hompage’, type: :feature do
    visit '/'
    expect(page).to have_content(‘Welcome to the cat website’)
  end
end

``` 
Cucumber is a way of writing tests to make them super-readable for to non-developers. For the above scenario, a test written in cucumber could look like:
```
Scenario: View homepage
	Given I am on the homepage
	Then I should see ‘Welcome to the cat website’
```
When you run the test, the lines Given and Then would then each run code in another file. This other file would have its code divided into ’steps’ - one for the Given line (to visit the homepage), and one for the Then line (to check the page content). It could look like something like this:
```
Given /^I am on the homepage$/ do
	visit root_path
end

Then /^I should see [regex for any text]$/ do |text|
	page.should have_content text
end
```

Note that the cucumber test uses capybara methods like `visit`.

Essentially, cucumber tests add an extra layer of code that is meant to be read by non-devs, like a stakeholder or product manager, so that they can understand what is being tested, and decide if we have met acceptance for a website.

Interestingly, if you look at the capybara example, it reads a lot like English anyway. And this is the aim of good tests written in Ruby - they should read like English. With this in mind, I am recommending we only use capybara and rspec for testing. Cucumber adds an unnecessary layer of code that won’t be read by anyone on our team, and since we are using Ruby, we will aim to write readable tests in any case.
